substrate/bin/utils/chain-spec-builder/README.md

```diff
@@ -0,0 +1,117 @@
+runtime function is called.
+
+### Display the runtime's default `GenesisConfig`
+
+```bash
+chain-spec-builder display-preset -r runtime.wasm
+```
+
+_Note:_ [GenesisBuilder::get_preset](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.get_preset)
+runtime function is called.
+
+### Display the `GenesisConfig` preset with given name
+
+```bash
+chain-spec-builder display-preset -r runtime.wasm -p "staging"
+```
+
+_Note:_ [GenesisBuilder::get_preset](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.get_preset)
+runtime function is called.
+
+### List the names of `GenesisConfig` presets provided by runtime
+
+```bash
+chain-spec-builder list-presets -r runtime.wasm
+```
+
+_Note:_ [GenesisBuilder::preset_names](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.preset_names)
+runtime function is called.
+
+### Generate chain spec using runtime provided genesis config preset
+
+Patch the runtime's default genesis config with the named preset provided by the runtime and generate the plain
+version of chain spec:
+
+```bash
+chain-spec-builder create -r runtime.wasm named-preset "staging"
+```
+
+_Note:_ [GenesisBuilder::get_preset](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.get_preset)
+and
+[GenesisBuilder::build_state](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.build_state)
+runtime functions are called.
+
+### Generate raw storage chain spec using genesis config patch
+
+Patch the runtime's default genesis config with provided `patch.json` and generate raw
+storage (`-s`) version of chain spec:
+
+```bash
+chain-spec-builder create -s -r runtime.wasm patch patch.json
+```
+
+_Note:_ [GenesisBuilder::build_state](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.build_state)
+runtime function is called.
+
+### Generate raw storage chain spec using full genesis config
+
+Build the chain spec using provided full genesis config json file. No defaults will be used:
+
+```bash
+chain-spec-builder create -s -r runtime.wasm full full-genesis-config.json
+```
+
+_Note_: [GenesisBuilder::build_state](https://docs.rs/sp-genesis-builder/latest/sp_genesis_builder/trait.GenesisBuilder.html#method.build_state)
+runtime function is called.
+
+### Generate human readable chain spec using provided genesis config patch
+
+```bash
+chain-spec-builder create -r runtime.wasm patch patch.json
+```
+
+### Generate human readable chain spec using provided full genesis config
+
+```bash
+chain-spec-builder create -r runtime.wasm full full-genesis-config.json
+```
+
+### Extra tools
+
+The `chain-spec-builder` provides also some extra utilities: `VerifyCmd`, `ConvertToRawCmd`,
```

Comment:
I think this page is really good, and generally much needed, given that `ChainSpecBuilder` and its related code is new. 

My nitpick for future would also be for you to consider only covering the most important user flows in the `README`, and leave the rest to `please see --help` for more. 

I am a fan of this principle as it reduces the maintenance of the README. for complex CLIs, copy-pasting the entire CLI spec into the readme will soon become a mess. It is more worthwhile to improve the documentation of the types in Rust that will lead to a more high quality `--help`. 

Feel free to incorporate this here as well, but I will leave it up to you to decide.

Reply:
I think I am good with how it is currently. 

At this point this `README.md` is the single source of truth for `chain-spec-builder`, with commands that are backed by integration tests. The `README.md` was created based on what we had in `lib.rs` as crate-level docs (which appear in `docs.rs` and `polkadot-sdk-docs`). I am thinking this can stay like this for now until a more opinionated view over it will be proposed.